osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/periodicmeterreads/AbstractPeriodicMeterReadsCommandExecutor.java

```diff
@@ -232,6 +228,21 @@ private AmrProfileStatusCodeDto readAmrProfileStatusCode(final DataObject amrPro
     return new AmrProfileStatusCodeDto(flags);
   }
 
+  protected boolean validateDateTime(
+      final Date meterReadTime, final Date beginDateTime, final Date endDateTime) {
+
+    if (meterReadTime.before(beginDateTime) || meterReadTime.after(endDateTime)) {
+      LOGGER.info(
+          "Not using an object from capture buffer (clock= {}), because the date does not match the given period: [ {} .. {} ].",
+          meterReadTime,
+          beginDateTime,
+          endDateTime);
+      return false;
+    } else {
+      return true;
+    }
+  }
+

```

Comment:
Why is this validation moved from the DlmsHelper to this AbstractCommandExecutor class?

Reply:
The validation is now on the resulting ResponseItems, so it's not related to DLMS and therefore the DlmsHelper is not the right place anymore.